### DISCUSSION TITLE
Backspace Control Character Not Working in Nested SSH Sessions

### DISCUSSION THREAD

**Discussion by prithvitewatia** [ghostty-org/ghostty#4153] [Q&A] [answered]
When using Ghostty Terminal, the Backspace control character (\b) does not work as expected when SSH-ing into multiple levels of servers. Also some other commands are not recognised  like
clear: 'xterm-ghostty': unknown terminal type.

 For example:

SSH into Server A from the local machine.
From Server A, SSH into Server B.
In the above scenario, the Backspace key fails to function correctly on Server B, making it difficult to edit commands.

Steps to Reproduce:

Open Ghostty Terminal.
SSH into a remote server (Server A).
From Server A, SSH into another server (Server B).
Try to use the Backspace key to delete characters in a command.

Expected Behavior:

The Backspace key should correctly delete characters at all levels of SSH sessions, including when nested.

Additional Information:

Ghostty Terminal version: 
Ghostty 1.0.0

Version
  - version: 1.0.0
  - channel: stable
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.none
  - font engine: font.main.Backend.coretext
  - renderer   : renderer.Metal
  - libxev     : main.Backend.kqueue
Operating System:
Mac OS sequoia 15.1.1
Server configurations:
uname -a Output
```txt
Server A: Linux <redacted> 5.4.0-1081-aws #88~18.04.1-Ubuntu SMP Thu Jun 23 16:29:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
Server B: Linux <redacted> 4.18.0-553.16.1.el8_10.x86_64 #1 SMP Thu Aug 8 08:05:28 PDT 2024 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by JakeWharton** [comment, marked as answer]
Hi! Please see https://ghostty.org/docs/help/terminfo#ssh

**Reply by marknefedov** [reply]
Is there a way to make Ghostty present itself with known terminfo? I need to work with remote ssh servers, where I do not have administrator rights, and I have no access to SSH configuration.

**Reply by jcollie** [reply]
Read the rest of the page that Jake linked and it shows how to set up SSH to do what you want. https://ghostty.org/docs/help/terminfo#configure-ssh-to-fall-back-to-a-known-terminfo-entry

**Comment by prithvitewatia** [comment]
Hi @JakeWharton Thanks. It helped. I have created a fall back to a known terminfo entry.
